src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java

```diff
@@ -115,6 +116,9 @@ public IntegrationResults integrateAssociations(
             @RequestBody Set<Association> associations,
             @RequestParam( value = DETAILED_RESULTS, required = false, defaultValue = "false" )
                     boolean detailedResults ) {
+        if ( associations.isEmpty() ) {
+            return new IntegrationResults( 0, 0, Optional.empty(), Optional.empty() );
```

Comment:
🤔 
Do we have a ticket to make shuttle not upload in this case?

Reply:
This was caused by some weirdness in case loader which calls data integration api. I double checked RestDestination in shuttle and upload is not executed if empty set of data is passed in.